iOS/Flows/Room/RoomViewController.swift

```diff
@@ -139,47 +143,77 @@ class RoomViewController: DiffableCollectionViewController<RoomSectionType,
         return data
     }
     
-    @MainActor
-    func reloadNotices() async {
-        
-        try? await NoticeStore.shared.initializeIfNeeded()
-        let notices = await NoticeStore.shared.getAllNotices()
-        
-        var items: [RoomItemType] = notices.compactMap({ notice in
-            return .notice(notice)
-        })
+    
+    private var loadAllTask: Task<Void, Never>?
+    
+    private func reloadAllSections() {
+        self.loadAllTask?.cancel()
         
-        let addItems: [RoomItemType] = PeopleStore.shared.unclaimedReservations.keys.compactMap { reservationId in
-            return .add(reservationId)
+        self.loadAllTask = Task { [weak self] in
+            guard let `self` = self else { return }
+            
+            self.reloadNotices()
```

Comment:
I decided to only reload the top sections as the conversations are driven by segment control. More work than its worth at this point.